src/Utility.js

```diff
@@ -42,7 +48,7 @@ export function test (value, pattern) {
  * @return {string?}
  */
 export function match (value, pattern) {
-	return pattern.exec(value)[0]
+	return test(value, pattern) ? pattern.exec(value)[0] : null
```

Comment:
im curious - are those optimized by engines so subsequent calls on the same inputs are cached? or is `exec` more heavy? any particular reason why this has been chosen instead of:
```js
return (value = pattern.exec(value)) ? value[0] : value
```

Reply:
That's probably better.

Reply:
My proposed version?

Reply:
Yes.

Reply:
K, I've pushed the change directly to this branch